galvan/src/main/java/org/terracotta/testing/config/ConfigRepositoriesBuilder.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.testing.config;
+
+import org.terracotta.ipceventbus.proc.AnyProcess;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.terracotta.testing.demos.TestHelpers.isWindows;
+
+public class ConfigRepositoriesBuilder {
+  private final Path kitOriginDir;
+  private final String clusterName;
+  private final Path[] tcConfigPaths;
+  private final Path testParentDir;
+
+  public ConfigRepositoriesBuilder(Path kitOriginDir, Path testParentDir, String clusterName, Path... tcConfigPaths) {
+    this.kitOriginDir = kitOriginDir;
+    this.testParentDir = testParentDir;
+    this.clusterName = clusterName;
+    this.tcConfigPaths = tcConfigPaths;
+  }
+
+  public Path build() {
+    List<String> command = new ArrayList<>();
+    String script = kitOriginDir.resolve("tools").resolve("config-convertor").resolve("bin").resolve("config-convertor") + (isWindows() ? ".bat" : ".sh");
```

Comment:
This is relying on a tool that lives only in tc-platform.  It would be nice if this class could live closer to the code that produces the tool.  Maybe the config builders can be moved to galvan support modules in tc-core and tc-platform.

Reply:
Here's the complete picture of the problem:

1. tc-core has no knowledge of dynamic-config. It also defines the galvan-support module. So tests in tc-core need to use tc-config + the old scripts.
2. tc-platform contains the dynamic-config code. It depends on galvan-support created by tc-core but not directly on galvan. Tests in tc-platform need to use tc-config (internally converted to a config repo) + the new startup scripts.
3. tc-ee creates it's own galvan-support-ee, and depends on galvan. It needs to use config repository + the new startup scripts.

Thus:
1. No changes related to dynamic-config can be made in galvan-support
2. Since tc-platform relies on galvan support, tc-platform is helpless.
3. tc-ee needs to use some of this conversion code.
4. Galvan should be able to use any kit we want it to run against.

With the above points in mind, keeping these files here made sense to me.

Reply:
Can we create a configuration builder interface that is passed into galvan?  That way galvan can be config agnostic and config can be properly passed in from tc-core, tc-platform or wherever.

Reply:
My thought is this is a migration away from a legacy configuration scheme.  As we move forward the old xml config will be less and less relevant and can be dropped all together.  Configuration is pluggable, the testing support for it should be pluggable as well.

Reply:
This seems like the right way to go about it, but isn't doable given the scenarios that we have to support. Galvan has the starter scripts, which need to know which configuration style (XML/properties/config repo etc.) is in use - so galvan can't be config agnostic in the true sense, as long as Galvan needs to support both the scripts.

An alternative could be to remove galvan-support from tc-core, and either move it to galvan or tc-platform. Most of the tests there are testing Galvan itself, and not tc-core. This would simplify a lot of things, and make Galvan much cleaner.

I've sent you a meeting invite to avoid the back-and-forth discussion on Github, and converge towards a conclusion at the earliest.

Reply:
PR updated based on review comments. Galvan is now agnostic of configs and startup scripts. Kept the config builders here though as they're needed at multiple places, and I didn't find any better location for these.